python_modules/libraries/dagster-components/dagster_components/core/component.py

```diff
@@ -362,11 +363,11 @@ def build_defs(self, context: ComponentLoadContext) -> Definitions:
         """
         abs_file_path = file_path.absolute()
         with pushd(str(self.path)):
-            abs_context_path = self.path.absolute()
             # Problematic
             # See https://linear.app/dagster-labs/issue/BUILD-736/highly-suspect-hardcoding-of-components-string-is-component-relative
-            component_module_relative_path = abs_context_path.parts[
-                abs_context_path.parts.index("defs") + 2 :
+            module_id = self.module_name.split(".")[-1]
```

Comment:
this is a slight improvement to the jank logic that existed here before, which no longer assumes that the top level module is named "defs". I still think this whole thing should be reworked but I was in this area while debugging some other stuff and felt that this made things marginally better